### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by AndrewHudzenko**

### CODE REVIEW

**Review by Toltbcm — APPROVED** [review]

File: src/main/java/practice/StreamPractice.java [hunk 1]
```diff
@@ -23,7 +35,12 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(n -> n % 2 != 0 ? numbers.get(n) - 1 : numbers.get(n))
```
Comment (Toltbcm) [inline src/main/java/practice/StreamPractice.java:39]:
```suggestion
                .map(i -> i % 2 != 0 ? numbers.get(i) - 1 : numbers.get(i))
```
I think it's better.
i - index

**Review by Toltbcm — CHANGES_REQUESTED** [review]

File: src/main/java/practice/StreamPractice.java (hunk 1 above)
Comment (AndrewHudzenko) [reply src/main/java/practice/StreamPractice.java:39]: fixed

**Review by Toltbcm — APPROVED** [review]

File: src/main/java/practice/StreamPractice.java [hunk 2]
```diff
@@ -23,7 +35,12 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(i -> i % 2 != 0 ? numbers.get(i) - 1 : numbers.get(i))
+                .filter(i -> i % 2 != 0)
+                .mapToDouble(i -> i)
+                .average()
+                .orElseThrow(NoSuchElementException::new);
```
Comment (Noct2000) [inline src/main/java/practice/StreamPractice.java:43]: add error message

File: src/main/java/practice/StreamPractice.java [hunk 3]
```diff
@@ -50,7 +71,11 @@ public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toA
      */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(person -> person.getSex().equals(Person.Sex.MAN)
+                        ? person.getAge() >= fromAge && person.getAge() <= maleToAge
+                        : person.getAge() >= fromAge && person.getAge() <= femaleToAge)
+                .collect(Collectors.toList());
```
Comment (Noct2000) [inline src/main/java/practice/StreamPractice.java:78]: you can use in the `.filter` only boolean expression, add `||` operators, and you can check `fromAge` only once

**Review by Noct2000 — CHANGES_REQUESTED** [review]

File: src/main/java/practice/StreamPractice.java [hunk 4]
```diff
@@ -14,7 +18,15 @@ public class StreamPractice {
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .map(s -> s.split(","))
+                .flatMap(Arrays::stream)
```
Comment (Noct2000) [inline src/main/java/practice/StreamPractice.java:23]: you can combine operations in the `.flatMap`

File: src/main/java/practice/StreamPractice.java [hunk 5]
```diff
@@ -14,7 +18,15 @@ public class StreamPractice {
...
+                .map(Integer::parseInt)
+                .mapToInt(n -> n)
```
Comment (Noct2000) [inline src/main/java/practice/StreamPractice.java:25]: you can use `Integer::parseInt` in the `.mapToInt`

**Review by Noct2000 — CHANGES_REQUESTED** [review]

**Review by Noct2000 — APPROVED** [review]

**Pull request closed without merging**